services/gundeck/test/integration/API.hs

```diff
@@ -259,6 +259,9 @@ sendMultipleUsers gu ca _ _ = do
             let p2 = fromJSON (Object (List1.head (n^.queuedNotificationPayload)))
             in assertEqual "Wrong events in notification" (Success (PushRemove tok)) p2
   where
+    checkNotifications []     = error "No notifications received!"
```

Comment:
this seems redundant, since there's already a `length >= 2` assertion above?

Reply:
Yes, but it causes a
```
wire-server/services/gundeck/test/integration/API.hs:253:13: warning: [-Wincomplete-uni-patterns]
    Pattern match(es) are non-exhaustive
    In a pattern binding: Patterns not matched: []
```
which would be nice to avoid imho

Reply:
ok, makes sense. I don't know if we get a better error message if we use `assertFail` instead of `error`, they are probably similar.

Reply:
Yeah, `error` is a bit more convenient in this case as `assertFailure` returns `IO ()`